product_dimension/models/product.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2015 ADHOC SA  (http://www.adhoc.com.ar)
+# Copyright 2015-2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models, fields, api
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    @api.onchange('length', 'height', 'width', 'dimensional_uom_id')
+    def onchange_calculate_volume(self):
+        if self.length and self.height and self.width and \
```

Comment:
these 2 methods are repeated into prod template: move them to simple functions and call them where needed

Reply:
I added your improvements except this one: 
1) these functions are already simple functions. 
one could, if it makes sense, encapsulate further and move the function on product.product and call it from product.template

2) the PR has already been approved by @leemannd

Reply:
@simahawk @leemannd 
done

Reply:
@leemannd @simahawk 
done

Reply:
Thanks